nscplugin/src/main/scala/scala/scalanative/nscplugin/NirGenExpr.scala

```diff
@@ -1549,15 +1471,20 @@ trait NirGenExpr { self: NirGenPhase =>
         case D2L => (nir.Type.Double, nir.Type.Long)
         case D2F => (nir.Type.Double, nir.Type.Float)
         case D2D => (nir.Type.Double, nir.Type.Double)
+
+        case CAST_RAWWORD_TO_INT => (nir.Type.Word, nir.Type.Int)
+        case CAST_RAWWORD_TO_LONG => (nir.Type.Word, nir.Type.Long)
+        case CAST_INT_TO_RAWWORD => (nir.Type.Int, nir.Type.Word)
+        case CAST_LONG_TO_RAWWORD => (nir.Type.Long, nir.Type.Word)
```

Comment:
Those will never be reached here. You need to put it in `genRawCastOp` instead, also make sure to modify `isRawCastOp` and make sure `CAST_` primitives are in the group as the other cast intrinsics.

Reply:
I actually set up `genApplyPrimitive` to just use `genCoercion` for raw cast ops, so the selection of cast types ends up coming here (verified by removing the cases and seeing that there is a `MatchError`).

Reply:
Lets not do that. We try to separate Scalac primitives, from our own. This makes it easier to reason about.